File: ruby_env/test/test_calculator.rb
require_relative 'test_helper'
require 'calculator'

# Tests for the Calculator class
class TestCalculator < Minitest::Test
  # Setup before each test method
  def setup
    @calc = Calculator.new
  end
  
  # Cleanup after each test method
  def teardown
    # Perform cleanup work here
  end
  
  # Test the add method
  def test_add
    assert_equal(3, @calc.add(1, 2))
    assert_equal(0, @calc.add(-1, 1))
    assert_equal(-2, @calc.add(-1, -1))
  end
  
  # Test the subtract method
  def test_subtract
    assert_equal(1, @calc.subtract(3, 2))
    assert_equal(0, @calc.subtract(1, 1))
    assert_equal(0, @calc.subtract(-1, -1))
  end
  
  # Test the multiply method
  def test_multiply
    assert_equal(6, @calc.multiply(2, 3))
    assert_equal(0, @calc.multiply(0, 5))
    assert_equal(-6, @calc.multiply(-2, 3))
  end
  
  # Test the divide method
  def test_divide
    assert_equal(2, @calc.divide(6, 3))
    assert_equal(2.5, @calc.divide(5, 2))
    assert_equal(-2, @calc.divide(-6, 3))
    
    # Test division by zero exception
    assert_raises(ZeroDivisionError) do
      @calc.divide(5, 0)
    end
  end
end

File: python_env/tests/test_calculator.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
测试计算器模块
"""

import unittest
from src.calculator import Calculator


class TestCalculator(unittest.TestCase):
    """
    测试计算器类的功能
    """

    def setUp(self):
        """
        每个测试方法执行前的设置
        """
        self.calc = Calculator()

    def test_add(self):
        """
        测试加法方法
        """
        self.assertEqual(self.calc.add(1, 2), 3)
        self.assertEqual(self.calc.add(-1, 1), 0)
        self.assertEqual(self.calc.add(-1, -1), -2)

    def test_subtract(self):
        """
        测试减法方法
        """
        self.assertEqual(self.calc.subtract(3, 2), 1)
        self.assertEqual(self.calc.subtract(1, 1), 0)
        self.assertEqual(self.calc.subtract(-1, -1), 0)

    def test_multiply(self):
        """
        测试乘法方法
        """
        self.assertEqual(self.calc.multiply(2, 3), 6)
        self.assertEqual(self.calc.multiply(0, 5), 0)
        self.assertEqual(self.calc.multiply(-2, 3), -6)

    def test_divide(self):
        """
        测试除法方法
        """
        self.assertEqual(self.calc.divide(6, 3), 2)
        self.assertEqual(self.calc.divide(5, 2), 2.5)
        self.assertEqual(self.calc.divide(-6, 3), -2)
        
        # 测试除以零的异常
        with self.assertRaises(ZeroDivisionError):
            self.calc.divide(5, 0)


if __name__ == '__main__':
    unittest.main()
